src/averageAges.js

```diff
@@ -37,7 +48,30 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter((person) => {
+    const isFemale = person.sex === 'f';
+    const isMother = people.some((child) => child.mother === person.name);
+
+    if (withChildren) {
+      return isFemale && isMother;
+    }
+
+    return isFemale;
+  });
+
+  if (!women.length) {
+    return 0;
+  }
```

Comment:
actually, you can move this check inside calculateAverageAge function

Reply:
Thx 😊 done ✅